report_thesis/src/sections/proposed_approach/optimization_framework.tex

```diff
@@ -0,0 +1,165 @@
+However, depending on the number of models in consideration, a high number of trials can quickly become computationally prohibitive with our approach.
+The $optimize$ function uses the number of trials and the objective of minimizing the \texttt{rmse\_cv} and \texttt{std\_dev\_cv} to manage the optimization process and mediating the metrics returned by the \nameref{alg:combined_objective} function to the sampler. 
+
+This leads us to the \nameref{alg:combined_objective} function which, as previously mentioned, returns the evaluation metrics for the given trial.
+
+To this function we supply the current model $m$, the Target Oxide $o$, and the sampler.
+
+In Lines~\ref{step:sample_hyperparameters} to~\ref{step:instantiate_dim_reduction} we instantiate the model and the preprocessors with the hyperparameters sampled by the sampler.
+The sampling process is being guided by the objective function, via the returned metrics from the previous trial.
+In our setup, we always require that a scaler is instantiated, to ensure that the data is correctly scaled.
+However, to measure the impact of data transformation and dimensionality reduction, we allow for these to be initialized as an identity function, which does not modify the data.
+Whether to instantiate a specific preprocessor or an identity function is determined by the sampler.
+
+Once the preprocessors are instantiated, we construct a pipeline of these to ensure that the data is processed in the order they are defined, seen in Line~\ref{step:construct_pipeline}.
+The order is important, as the preprocessing steps are to a degree interdependent: scaling ensures that all features are on a similar range, which is ensures that transformations are applied uniformly.
+Similarly, dimensionality reduction techniques typically also produce better results if the data has been scaled.
+However, it may not always be advantageous or yield better results if the data has already been transformed.
+As such, we allow for the optimization framework to optionally use these if they are deemed to be beneficial.
+
+In Line~\ref{step:get_data} to~\ref{step:apply_pipeline} we fetch the data, apply our data partitioning strategy to generate four cross-validation sets, a training set and a test set, and apply the preprocessing to the datasets.
+The purpose of fetching the data for each trial is to ensure no modifications leak through trials, corrupting the dataset over time.
+This prevents any form of double preprocessing from occuring, leading to potential issues.
+
+As mentioned in Section~\ref{subsec:validation_testing_procedures}, we use both cross validation and a test set to evaluate the model.
+This can be seen in line Line~\ref{step:cross_validate} and Lines~\ref{step:train_model} to~\ref{step:evaluate_model}, where cross validation, training, and evaluation are performed with respect to the current oxide.
```

Comment:
```suggestion
This can be seen in line Line~\ref{step:cross_validate} and Lines~\ref{step:train_model} to~\ref{step:evaluate_model}, where cross-validation, training, and evaluation are performed with respect to the current oxide.
```